Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -190,6 +169,64 @@ public static string GetLanguageVersion(string targetFramework)
             return Unknown();
         }
 
+        // Knowledge on unique machine Ids for different OSes obtained from https://github.com/denisbrodbeck/machineid
+        public static string GetAnonymizedUserId()
+        {
+            var id = string.Empty;
+            var currentOs = System.Environment.OSVersion.Platform;
+            Match match = null;
+            try
+            {
+                if (currentOs == PlatformID.Win32S || currentOs == PlatformID.Win32Windows ||
+                    currentOs == PlatformID.Win32NT || currentOs == PlatformID.WinCE)
+                {
+                    var machineIdToParse = RunProcess("reg", "QUERY HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Cryptography -v MachineGuid");
+                    var regex = new Regex("^\\s+MachineGuid\\s+\\w+\\s+((\\w+-?)+)", RegexOptions.Multiline);
+                    match = regex.Match(machineIdToParse);
+                }
+                else if (currentOs == PlatformID.MacOSX)
+                {
+                    var machineIdToParse = RunProcess("ioreg", "-rd1 -c IOPlatformExpertDevice");
+                    var regex = new Regex("^.*IOPlatformUUID\\\"\\s=\\s\\\"(.+)\\\"", RegexOptions.Multiline);
```

Comment:
The RegExp used in dart is `"IOPlatformUUID" = "([^"]*)"`. Any reason not to use the same one?

Reply:
No, I just didn't look at other implementations. But we basically do the same. We look for some `IOPlatformUUID = "` and grab all it's in between the first and last apex.
Do we have to align on these things too? Like even regexes have to be reported in the scope doc?

Reply:
I would prefer if we do as using different regex-es can mean we could end up parsing different ids for the same machine.

Reply:
"Alignment notes" of the scope docs updated.